src/ddo/DDO.ts

```diff
@@ -212,18 +272,44 @@ export class DDO {
     this.proof = await this.generateProof(publicKey)
   }
 
-  public async addService(service: any): Promise<void> {
+  /**
+   * It reorders the services of the DDO using the service index
+   */
+  public reorderServices(): void {
+    this.service.sort((a, b) => (a.index > b.index ? 1 : -1))
+    for (let i = 0; i < this.service.length; i++) {
+      this.service[i].index = i
+    }
+    this.service.sort((a, b) => (a.index > b.index ? 1 : -1))
+  }
+
+  /**
+   * Adds a service to the DDO.
+   * @param service
+   */
+  public addService(service: ServiceCommon) {
     this.service.push(service)
   }
 
-  public async replaceService(index: number, service: any): Promise<void> {
+  /**
+   * Replaces a service in the DDO.
+   * @param index
+   * @param service
+   */
+  public replaceService(index: number, service: any) {
```

Comment:
should we check first if the index exists?

Reply:
:+1: